File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        nums.mapIndexed { index, num ->
            if (target - num in map) return intArrayOf(map.getValue(target - num), index)
            map.put(num, index)
        }
        return IntArray(0)
    }
}

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
	i := 0
	j := len(nums) - 1
	for {
		if i == j {
            if j == 0 {
                return []int{}
            }

            j--
            i = 0
		}

		if nums[i]+nums[j] == target {
			return []int{i, j}
		}

        i++ 
	}
}